packages/glimmer-runtime/lib/compiled/opcodes/content.ts

```diff
@@ -27,12 +39,20 @@ export class AppendOpcode extends Opcode {
   evaluate(vm: VM) {
     let reference = vm.frame.getOperand();
 
-    let mapped = map(reference, normalizeTextValue);
+    let mapped = map(reference, normalizeTextOrTrustedValue);
     let cache = new ReferenceCache(mapped);
-    let node = vm.stack().appendText(cache.peek());
-
-    if (!isConst(reference)) {
-      vm.updateWith(new UpdateAppendOpcode(cache, node));
+    let value = cache.peek();
+
+    if (isSafeString(value)) {
+      let bounds = vm.stack().insertHTMLBefore(null, value);
```

Comment:
I still think there is an issue with const detection in this case. Just because the input to a helper is const doesn't mean the ouput is -- even if it's a simple helper. The only way to guarantee that would be to assume that all simple helpers are pure and I don't think that's a safe assumption.

What if I implement a helper like:

```function random([min, max]) {
    return Math.floor(Math.random() * (max - min + 1) + min);
}```

```<input value={{random 1 10}}/>```

The inputs are const but the output isn't. Will there be a way for developers to specify their constness?